src/java.desktop/windows/classes/sun/awt/shell/Win32ShellFolder2.java

```diff
@@ -1437,7 +1441,7 @@ public Image getResolutionVariant(double width, double height) {
                     }
                 }
             }
-            if (retVal.getWidth(null) != w) {
+            if ((retVal != null) && (retVal.getWidth(null) != w)) {
```

Comment:
I believe this change is not needed.

Reply:
> You may still add `assert` statements into `MultiResolutionIconImage` constructors which ensures `resolutionVariants` contains no `null` values.

Will add assert statements in this `public MultiResolutionIconImage(int baseSize, Image image)` as null checks are handled in other constructor.

Reply:
> > You may still add `assert` statements into `MultiResolutionIconImage` constructors which ensures `resolutionVariants` contains no `null` values.
> 
> Will add assert statements in this `public MultiResolutionIconImage(int baseSize, Image image)` as null checks are handled in other constructor.

What do you mean?

There are no `null`-checks in the other constructor. Or do I miss anything?

It's just you don't call the other constructor any more if an icon fetched from the OS is `null`. (Perhaps, it's not the most effective way: if another icon size is extracted correctly, a non-null value could still be returned.)

The following statement

```java
assert !resolutionVariants.containsValue(null) : "There are null icons in the variants map";
```

would still prevent a `MultiResolutionIconImage` from being created if any resolution variant contains `null`. The code above could change again and we may end up with the same problem again.

Reply:
Means the other constructor is called in two places only and in those places null checks are done. So, even if we add assertion inside constructor, it will unreachable.

Reply:
> Means the other constructor is called in two places only and in those places null checks are done. So, even if we add assertion inside constructor, it will unreachable.

It won't be unreachable, _it will prevent such a situation from happening again_. The code which calls the other constructor could be changed.

I still think it's reasonable to add the assertion. It's a no-op if assertions aren't enabled (the default), and it'll catch the broken invariant in testing since tests are (usually) run with assertions enabled.

Reply:
Yeah, got your point. It's better to add as a preventive measure.